src/index.test.js

```diff
@@ -244,16 +244,60 @@ describe("prepass", () => {
   });
 
   describe("hooks", () => {
-    it("it should support useState", async () => {
-      let setStateHoisted;
+    it("should not enqueue components for re-rendering when using setState", async () => {
+      let didUpdate = false;
+
+      // a re-render would invoke an array sort on the render queue, thus lets check nothing does so
+      const arraySort = jest.spyOn(Array.prototype, "sort");
```

Comment:
Super hacky to spy on `Array.prototype.sort` and it breaks when another test fails as jest will use `Array.prototype.sort` to format the diff for the other test. I couldn't come up with a different approach to make sure that we do not enqueue a re-render in core as everything is scoped private.